app/app/settings.py

```diff
@@ -166,13 +166,22 @@
 OSU_CLIENT_ID = int(os.getenv("OSU_CLIENT_ID"))
 OSU_CLIENT_SECRET = os.getenv("OSU_CLIENT_SECRET")
 OSU_REDIRECT_URL = os.getenv("OSU_REDIRECT_URL")
-OSU_LOGIN_URL = AuthHandler(
+OSU_DEV_SERVER = bool(int(os.getenv("OSU_DEV_SERVER")))
+
+AUTH = AuthHandler(
     OSU_CLIENT_ID,
     OSU_CLIENT_SECRET,
     OSU_REDIRECT_URL,
     Scope.identify()
-).get_auth_url()
-OSU_CLIENT = Client.from_client_credentials(OSU_CLIENT_ID, OSU_CLIENT_SECRET, OSU_REDIRECT_URL)
+)
+if OSU_DEV_SERVER:
+    AUTH.set_domain("dev.ppy.sh")
+OSU_LOGIN_URL = AUTH.get_auth_url()
+
+CLIENT = Client.from_client_credentials(OSU_CLIENT_ID, OSU_CLIENT_SECRET, OSU_REDIRECT_URL)
+if OSU_DEV_SERVER:
+    CLIENT.set_domain("dev.ppy.sh")
```

Comment:
I think this actually won't work since `set_domain` doesn't change the auth domain (though it should). You'll just have to do `CLIENT.auth.set_domain` to work around that.

Reply:
changed